### PR TITLE
tidb: load and apply common global variables for the session

### PR DESCRIPTION
**Pull request by coocood**
Fixes a bug some global variables like 'sql_mode' not applied to session.

### CODE REVIEW

**Comment by coocood** [comment]
@shenli @hanfei1991 PTAL

**Comment by shenli** [comment]
LGTM
@zimulala @hanfei1991 PTAL

File: sessionctx/variable/session.go [hunk 1]
```diff
@@ -244,18 +253,22 @@ func (s *SessionVars) SetSystemVar(key string, value types.Datum) error {
...
 			s.StrictSQLMode = false
 		}
-	} else if key == TiDBSnapshot {
+	case TiDBSnapshot:
 		err = s.setSnapshotTS(sVal)
 		if err != nil {
 			return errors.Trace(err)
 		}
+	case AutocommitVar:
+		isAutocommit := sVal == "ON" || sVal == "on" || sVal == "1"
```
Comment (hanfei1991) [inline sessionctx/variable/session.go]: How about "On"? Should you cast sVal to lower case then check it?

File: session.go [hunk 2]
```diff
@@ -811,3 +784,45 @@ func finishBootstrap(store kv.Storage) {
 		log.Fatalf("finish bootstrap err %v", err)
 	}
 }
+
+const loadCommonGlobalVarsSQL = "select * from mysql.global_variables where variable_name in ('" +
```
Comment (hanfei1991) [inline session.go]: There are two questions:
1. How do you write global vars to this table?
2. How about to set this table_name to a const string?

**Comment by zimulala** [comment]
Rest LGTM

File: session.go (hunk 2 above)
Comment (coocood) [inline session.go]: 1. By executing update statement on this table.
2. Then there will be too many constant strings to concat, hard to read.

**Comment by hanfei1991** [comment]
LGTM

**Pull request merged**